flink-runtime/src/main/java/org/apache/flink/runtime/taskexecutor/slot/TaskSlotTableImpl.java

```diff
@@ -523,6 +545,103 @@ public JobID getOwningJob(AllocationID allocationId) {
         }
     }
 
+    private void tryPersistAllocationSnapshot(
+            int requestedIndex,
+            JobID jobId,
+            AllocationID allocationId,
+            ResourceProfile resourceProfile) {
+        if (!allocationsDirectory.exists() && !allocationsDirectory.mkdirs()) {
+            LOG.debug(
+                    "Allocations directory {} doesn't exist and cannot be created.",
+                    allocationsDirectory.toPath());
```

Comment:
Then we should probably fail. I would also try to set these things up when we start the `TaskSlotTable`.